### ISSUE TITLE
Aggregation on non-lowercase column returns incorrect result in Pinot connector

### ISSUE THREAD

**Issue by mapshen** [trinodb/trino#10148]
When query Pinot via Trino (version 362), the `avg()` function doesn’t seem to work correctly. It always returns no data…Cannot even get it to work with dynamic tables.

**Comment by ebyhr** [comment]
What Pinot version are you using? Can you try upstream Trino (365-SNAPSHOT)? It has some enhancements comparing to 362.

**Comment by mapshen** [comment]
> What Pinot version are you using? Can you try upstream Trino (365-SNAPSHOT)? It has some enhancements comparing to 362.

362. I did look at the release notes for 365 but didn't see fixes related to this issue?

**Comment by ebyhr** [comment]
I asked Pinot version (not Trino). #9098 has various changes.

**Comment by mapshen** [comment]
@ebyhr tried 365 with Pinot 0.8.0 and 0.9.0, and `avg()` still doesn't work.

It seems to be due to the letter case...

When I issue query 
```
select max(A), avg(A) from <table> group by B
```
Trino translates it into
```
select max(A), avg(a) from <table> group by B
```
whereas Pinot can only recognize column `A`, not column `a`.

**Comment by ebyhr** [comment]
Thanks for sharing the example, I could reproduce it.

**Comment by elonazoulay** [comment]
So the table has 2 columns where they are equal if the case is ignored?

**Comment by mapshen** [comment]
> So the table has 2 columns where they are equal if the case is ignored?

column `A` and column `a` are supposed to be the same column. All columns in Pinot should be uppercase if I am not mistaken. The problem is Trino lowercases instead of uppercasing (?) the column name passed into `avg()`.

**Comment by elonazoulay** [comment]
Also found that `approx_distinct` is affected by this. Added tests to verify the fix in #9781.

**Comment by hashhar** [comment]
Fixed in https://github.com/trinodb/trino/pull/9781
